automation/jleechanorg_pr_automation/automation_utils.py

```
@@ -247,21 +242,57 @@ def execute_subprocess_with_timeout(cls, command: list, timeout: int = None,
+                temp_path = temp_file.name
+
+            # Set restrictive permissions before moving
+            os.chmod(temp_path, 0o600)  # Owner read/write only
+
+            # Atomic rename - this operation is atomic on POSIX systems
+            os.rename(temp_path, file_path)
+            temp_path = None  # Successful, don't clean up
+
+        except (OSError, json.JSONEncodeError) as e:
+            # Clean up temp file on error
+            if temp_path and os.path.exists(temp_path):
```

Ruff (0.14.11)

`os.path.exists()` should be replaced by `Path.exists()`

(PTH110)